commit 66dca9b8c50b5e59d3bea8b21cee5c6dae6c9c46
Author: Joe Malicki <jmalicki@metacarta.com>
Date:   Thu Aug 14 17:14:48 2008 -0400

    [SCSI] megaraid_sas: remove sysfs dbg_lvl world writeable permissions
    
    /sys/bus/pci/drivers/megaraid_sas/dbg_lvl defaults to being
    world-writable, which seems bad (letting any user affect kernel driver
    behavior and logging level).
    
    This turns off group and user write permissions, so that on typical
    production systems only root can write to it.
    
    [jejb: fix up rejections]
    Signed-off-by: Joseph Malicki <jmalicki@metacarta.com>
    Acked-by: "Yang, Bo" <Bo.Yang@lsi.com>
    Signed-off-by: James Bottomley <James.Bottomley@HansenPartnership.com>

diff --git a/drivers/scsi/megaraid/megaraid_sas.c b/drivers/scsi/megaraid/megaraid_sas.c
index 870dc1c..97b7633 100644
--- a/drivers/scsi/megaraid/megaraid_sas.c
+++ b/drivers/scsi/megaraid/megaraid_sas.c
@@ -3405,7 +3405,7 @@ megasas_sysfs_set_dbg_lvl(struct device_driver *dd, const char *buf, size_t coun
 	return retval;
 }
 
-static DRIVER_ATTR(dbg_lvl, S_IRUGO|S_IWUGO, megasas_sysfs_show_dbg_lvl,
+static DRIVER_ATTR(dbg_lvl, S_IRUGO|S_IWUSR, megasas_sysfs_show_dbg_lvl,
 		megasas_sysfs_set_dbg_lvl);
 
 static ssize_t
